tests/Doctrine/Tests/Common/Cache/FileCacheTest.php

```diff
@@ -27,19 +27,19 @@ public function getProviderFileName()
     {
          return array(
             //The characters :\/<>"*?| are not valid in Windows filenames.
-            array('key:1', 'key-1'),
-            array('key\2', 'key-2'),
-            array('key/3', 'key-3'),
-            array('key<4', 'key-4'),
-            array('key>5', 'key-5'),
-            array('key"6', 'key-6'),
-            array('key*7', 'key-7'),
-            array('key?8', 'key-8'),
-            array('key|9', 'key-9'),
-            array('key[10]', 'key[10]'),
-            array('keyä11', 'key--11'),
-            array('../key12', '---key12'),
-            array('key-13', 'key__13'),
+            array('key:1', 'key%3A1'),
+            array('key\2', 'key%5C2'),
+            array('key/3', 'key%2F3'),
+            array('key<4', 'key%3C4'),
+            array('key>5', 'key%3E5'),
+            array('key"6', 'key%226'),
+            array('key*7', 'key%2A7'),
+            array('key?8', 'key%3F8'),
+            array('key|9', 'key%7C9'),
+            array('key[10]', 'key%5B10%5D'),
+            array('keyä11', 'key%C3%A411'),
+            array('../key12', '..%2Fkey12'),
```

Comment:
I'm mostly worried about directory traversal. While `.` is a perfectly valid file name, we may have crashes caused by clients producing cache keys that are just `.` or `..` (not allowed, afaik)